modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/schema.scala

```diff
@@ -11,19 +11,42 @@ object schema {
 
   case class GenericMetadata()
 
+  @derive(eqv, show)
+  sealed trait SignatureType extends EnumEntry
+
+  object SignatureType extends Enum[SignatureType] with SignatureTypeNamer {
+    val values = findValues
+
+    case object Ecdsa extends SignatureType
+    case object EcdsaRecovery extends SignatureType
+    case object Ed25519 extends SignatureType
+    case object Schnorr1 extends SignatureType
+    case object SchnorrPoseidon extends SignatureType
+  }
+
+  trait SignatureTypeNamer {
+    implicit val name: Encoder[SignatureType] =
+      Encoder.forProduct1("status")(_ match {
+        case SignatureType.EcdsaRecovery   => "ecdsa_recovery"
+        case SignatureType.Schnorr1        => "schnorr_1"
+        case SignatureType.SchnorrPoseidon => "schnorr_poseidon"
+        case signatureType                 => signatureType.toString.toLowerCase
+      })
+  }
+
   @derive(eqv, show)
   sealed trait ChainObjectStatus extends EnumEntry
 
-  object ChainObjectStatus extends Enum[ChainObjectStatus] with ChainObjectStatusCodecs {
+  object ChainObjectStatus extends Enum[ChainObjectStatus] with ChainObjectStatusNamer {
     case object Unknown extends ChainObjectStatus
     case object Pending extends ChainObjectStatus
     case object Accepted extends ChainObjectStatus
 
     override def values: IndexedSeq[ChainObjectStatus] = findValues
   }
 
-  trait ChainObjectStatusCodecs {
-    implicit val encode: Encoder[ChainObjectStatus] =
+  trait ChainObjectStatusNamer {
```

Comment:
What is *Namer and where did you find it? We don't have such a thing within the whole repository. It's encoder, decoder or codec.

Reply:
Renamed to `*Encoder`.